h2o-genmodel/src/main/java/water/genmodel/IGeneratedModel.java

```diff
@@ -54,6 +61,9 @@
     /** Returns domain values for all columns including response column. */
     public String[][] getDomainValues();
 
+    /** Returns original domain values for all columns including response column. */
+    public String[][] getOrigDomainValues();
```

Comment:
I don't understand why this new method is introduced and then it always returns null in `GenModel`.

At least in some cases, could `getOrigDomainValues` return the same values as `getDomainValues` ?

Reply:
this method is only useful when the categorical encoding is not handled by the model (!= AUTO), this is the case when we need the original domain and the original column names - it is overridden in the generated POJO

imho this mimics the behavior of model._output._origDomains - they are only set in certain scenarios an are null in others